quic/s2n-quic-tls/src/client.rs

```diff
@@ -142,6 +142,13 @@ impl Builder {
         Ok(self)
     }
 
+    /// Optionally, use Optional or None on the Client builder. The default auth type
+    /// without calling this method is ClientAuthType::Required
+    pub fn with_client_auth_type(mut self, auth_type: ClientAuthType) -> Result<Self, Error> {
```

Comment:
This is a server configuration, not a client: https://github.com/aws/s2n-tls/blob/6881358a394cc8620d3bb05698948e65e5612e68/api/s2n.h#L2075

```c
/**
 * Sets whether or not a Client Certificate should be required to complete the TLS Connection. 
 *
 * If this is set to `S2N_CERT_AUTH_OPTIONAL` the server will request a client certificate but allow the client to not provide one.
 * Rejecting a client certificate when using `S2N_CERT_AUTH_OPTIONAL` will terminate the handshake.
 */
```

Also, I would prefer we just copy the documentation to keep it consistent.

Reply:
The Client is always going to verify the Server's cert and authenticate the Client. This setting controls the Server behavior; if it should require authenticate or not. So as Cameron pointed out this should be exposed in server.rs instead. Here is a great article which explains mTLS https://www.cloudflare.com/learning/access-management/what-is-mutual-tls/